crates/fluvio-cluster/src/start/k8.rs

```diff
@@ -630,41 +637,62 @@ impl ClusterInstaller {
 
         self.install_app().await?;
 
-        let namespace = &self.config.namespace;
-
-        let pb = self.pb_factory.create();
-        pb.set_message("🔎 Discovering Fluvio SC");
         // before we do let's try make sure SPU are installed.
         check_crd(self.kube_client.clone())
             .await
             .map_err(K8InstallError::from)?;
 
-        let (host_name, port) = self.discover_sc_address(namespace).await?;
-
-        let address = format!("{}:{}", host_name, port);
-
-        let cluster_config =
-            FluvioConfig::new(address.clone()).with_tls(self.config.client_tls_policy.clone());
+        let sc_service = self.discover_sc_service(&self.config.namespace).await?;
+        let (external_host, external_port) =
+            self.discover_sc_external_host_and_port(&sc_service).await?;
+        let external_host_and_port = format!("{}:{}", external_host, external_port);
+
+        let (install_host_and_port, pf_process) = if self.config.use_k8_port_forwarding {
+            info!("Using K8 port forwarding for install");
+            let (install_host, install_port, pf_process) =
+                self.start_sc_port_forwarding(&sc_service).await?;
+            (
+                format!("{}:{}", install_host, install_port),
+                Some(pf_process),
+            )
+        } else {
+            (external_host_and_port.clone(), None)
+        };
 
+        let cluster_config = FluvioConfig::new(install_host_and_port.clone())
+            .with_tls(self.config.client_tls_policy.clone());
+        let pb = self.pb_factory.create();
+        pb.set_message("🔎 Discovering Fluvio SC");
         let fluvio =
             match try_connect_to_sc(&cluster_config, &self.config.platform_version, &pb).await {
                 Some(fluvio) => fluvio,
                 None => return Err(K8InstallError::SCServiceTimeout),
             };
-        pb.println(format!("✅ Connected to SC: {}", address));
+        pb.println(format!("✅ Connected to SC: {}", install_host_and_port));
         pb.finish_and_clear();
         drop(pb);
 
-        if self.config.save_profile {
-            self.update_profile(&address)?;
-        }
-
         // Create a managed SPU cluster
         self.create_managed_spu_group(&fluvio).await?;
+
```

Comment:
The profile must be saved before the SPU group is provisioned.  Otherwise, if SPU provisioning fails, there is no way to talk to the cluster.

Reply:
done